servers/src/main/java/tachyon/worker/block/BlockWorker.java

```diff
@@ -96,12 +96,12 @@ public BlockWorker(TachyonConf tachyonConf) throws IOException {
     // Setup metrics collection
     WorkerSource workerSource = new WorkerSource();
     mWorkerMetricsSystem = new MetricsSystem("worker", mTachyonConf);
-    workerSource.registerGauges(this);
-    mWorkerMetricsSystem.registerSource(workerSource);
-
     // Set up BlockDataManager
     mBlockDataManager = new BlockDataManager(tachyonConf, workerSource);
 
+    workerSource.registerGauges(mBlockDataManager);
```

Comment:
Please also move line 96,97,98 down, to the line right before `workerSource.registerGauges(mBlockDataManager);`, as they all belong to "Setup metrics collection" and we don't want to separate them